server/src/main/java/io/crate/expression/scalar/AgeFunction.java

```diff
@@ -112,13 +118,25 @@ public Period evaluate(TransactionContext txnCtx, NodeContext nodeContext, Input
      */
     private static Period getPeriod(long timestamp1, long timestamp2) {
         /*
-         Normalize is important as it affects the internal representation of the Period object.
+         PeriodType is important as it affects the internal representation of the Period object.
          PeriodType.yearMonthDayTime() is needed to return 8 days but not 1 week 1 day.
          Streamer of the IntervalType will simply put 0 in 'out.writeVInt(p.getWeeks())' as getWeeks() returns zero for unused fields.
          */
+
+
+        var id = availableIds.get(random.nextInt(availableIds.size()));
+        // System.out.println("random-obj-id:"+random.hashCode()+", id = " + id);
+        DateTimeZone.setDefault(DateTimeZone.forID(id));
```

Comment:
1. `TestRuleSetupAndRestoreClassEnv` from Lucene's package [set's timezone](https://github.com/apache/lucene/blob/b0d6fe68d1f2b8a93c9fc22a6ccdedff94bf1fbb/lucene/test-framework/src/java/org/apache/lucene/tests/util/TestRuleSetupAndRestoreClassEnv.java#L221) and locale randomly. In Jenkins [report](https://jenkins.crate.io/job/CrateDB/job/crate_on_pr/16771/testReport/junit/io.crate.expression.scalar/AgeFunctionTest/test_2_arguments_positive/) I saw a message `"All tests run in this JVM:"` and it reminded me about recently seen [SO post](https://stackoverflow.com/q/29533632/2104560) about code setting default timezone in runtime and affecting another code which used default time zone. 

2. I thought that it might be an issue with parallel run of tests where one test sets random timezone and another test gets affected. To check this idea without running the whole huge test pack multiple times I imitated this by randomly setting timezone right before Interval computation, and Interval is computed with default timezone (no timezone argument). After this change running 
` ./gradlew :server:test -Dtests.class=io.crate.expression.scalar.AgeFunctionTest -Dtests.method="test_2_arguments_positive"`  with this test having multiple `@Repeat` it started to fail in ˜10% of cases. 

3. After explicitly using UTC in Interval the test case doesn't fail even with 1000 repetitions (see 2 second commit)

4. We don't set default timezone in production code (at least I haven't found) - so it's not a fix but it wouldn't hurt anyway - not only to address flakiness but to be future proof if we by occasion set default timezone somewhere beside tests.

Reply:
Hi @seut, please take a look ^

Reply:
Ah ok, good catch! 👍